app/controllers/application_controller.rb

```diff
@@ -21,6 +23,21 @@ def configure_permitted_parameters
     ActionController::Base.helpers.number_to_percentage(account.balance / objective.amount * 100, precision: 0)
   end
 
+  helper_method memoize def topup_utc_time
+    schedule = top_up_schedule
+    Time.now.utc.change(hour: schedule[:hour], min: schedule[:min])
+  end
+
+  def top_up_schedule
```

Comment:
This is too complicated... WDYT if we just hardcode it?

I mean, yes there is a possibility of getting desynced value. But that probability is 0 because we are not going to change the schedule. On the other hand, if we introduce a dynamic schedule this code will be gone. So we should not write it now. It's just a dead code in the beginning. [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)

Reply:
I'm really looking forward to the implementation of a basic dynamic schedule, but I'm waiting for the results of the current approach. We acknowledge this, and it's also part of the learning experience.